File: ThirdYear/harshi706_HarshiAgrawal_2125it1200_3/Week_3/countAndSay.java
class Solution {
    public String countAndSay(int n) {
         if(n==1){
            return "1";
        }
        int count=0;
//Strings are immutable thats why using StringBuilder
        StringBuilder res=new StringBuilder();
        String s=countAndSay(n-1);
       
        for(int i=0;i<s.length();i++){
            count++;
            if(i==s.length()-1 || s.charAt(i)!=s.charAt(i+1)){
            res=res.append(count).append(s.charAt(i));
                count=0;
            }
        }
        return res.toString();
    }
}

File: ThirdYear/prabhat-04_PrabhatKeshari_2125it1086_3/Week_3/CountAndSay.cpp
class Solution {
public:
    string countAndSay(int n) {
        if(n==1)
            return "1";
        string prev = countAndSay(n-1);
        char curr = prev[0];
        int cnt=1;
        string ans = "";
        for(int i=1;i<prev.size();i++){
            if(prev[i]!=curr){
                ans += to_string(cnt) + curr;
                cnt=1;
                curr = prev[i];
            }
            else
                cnt++;
        }
        ans += to_string(cnt) + curr;
        return ans;
    }
};